cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -697,6 +699,100 @@ __global__ void __launch_bounds__(delta_preproc_block_size) gpuComputeDeltaPageS
   }
 }
 
+/**
+ * @brief Kernel for computing string page output size information for DELTA_LENGTH_BYTE_ARRAY
+ * encoding.
+ *
+ * This call ignores columns that are not DELTA_LENGTH_BYTE_ARRAY encoded. On exit the `str_bytes`
+ * field of the `PageInfo` struct will be populated.
+ *
+ * Currently this function only supports being called by a single warp.
+ *
+ * @param pages All pages to be decoded
+ * @param chunks All chunks to be decoded
+ * @param min_rows crop all rows below min_row
+ * @param num_rows Maximum number of rows to read
+ */
+__global__ void __launch_bounds__(delta_length_block_size) gpuComputeDeltaLengthPageStringSizes(
+  PageInfo* pages, device_span<ColumnChunkDesc const> chunks, size_t min_row, size_t num_rows)
+{
+  using cudf::detail::warp_size;
+  using WarpReduce = cub::WarpReduce<uleb128_t>;
+  __shared__ typename WarpReduce::TempStorage temp_storage;
+  __shared__ __align__(16) page_state_s state_g;
+  __shared__ __align__(16) delta_binary_decoder string_lengths;
+
+  page_state_s* const s = &state_g;
+  int const page_idx    = blockIdx.x;
+  int const t           = threadIdx.x;
+  PageInfo* const pp    = &pages[page_idx];
+
+  // whether or not we have repetition levels (lists)
+  bool const has_repetition = chunks[pp->chunk_idx].max_level[level_type::REPETITION] > 0;
+
+  // setup page info
+  if (!setupLocalPageInfo(
+        s, pp, chunks, min_row, num_rows, mask_filter{decode_kernel_mask::DELTA_LENGTH_BA}, true)) {
+    return;
+  }
+
+  bool const is_bounds_pg = is_bounds_page(s, min_row, num_rows, has_repetition);
+
+  // for DELTA_LENGTH_BYTE_ARRAY, string size is page_data_size - size_of_delta_binary_block.
+  // so all we need to do is skip the encoded string size info and then do pointer arithmetic,
+  // if this isn't a bounds page.
+  if (not is_bounds_pg) {
+    if (t == 0) {
+      auto const* string_start = string_lengths.find_end_of_block(s->data_start, s->data_end);
+      size_t len               = static_cast<size_t>(s->data_end - string_start);
+      // TODO check for overflow
```

Comment:
Might be easier to do this on on the host side since we bring the sizes back. Can do a quick check before allocating the char data for the column buffer.